frontend/src/pages/notebookController/screens/server/NotebookServer.tsx

```diff
@@ -35,8 +42,33 @@ const NotebookServer: React.FC = () => {
     [requestNotebookRefresh, navigate],
   );
 
+  const handleStopWorkbenches = React.useCallback(() => {
+    setIsDeleting(true);
+    stopWorkbenches(notebooksToStop, isAdmin)
+      .then(() => {
+        setIsDeleting(false);
+        onNotebooksStop(true);
+        setShowModal(false);
+      })
+      .catch((e) => {
+        setIsDeleting(false);
+        notification.error(
+          `Error stopping workbench${notebooksToStop.length > 1 ? 's' : ''}`,
+          e.message,
+        );
+      });
+  }, [isAdmin, notebooksToStop, notification, onNotebooksStop]);
+
   const link = currentUserNotebookLink || '#';
 
+  React.useEffect(() => {
+    if (notebooksToStop.length && !showModal && dontShowModalValue) {
+      handleStopWorkbenches();
+    } else if (notebooksToStop.length && !dontShowModalValue) {
+      setShowModal(true);
+    }
+  }, [notebooksToStop.length, showModal, handleStopWorkbenches, dontShowModalValue]);
```

Comment:
There are something wrong with this hook, I played around it for a while and I don't know when I triggered all workbenches stopped from the admin panel without seeing the modal, while I didn't hide the modal. Better to check it again to see how we can organize the modal show status and the checkbox status better. I can take a closer look if you need help.

Note: `useStopNotebookModalAvailability` hook can be used in different components, no need to pass it around.